core/trino-main/src/main/java/io/trino/operator/PartitionedOutputOperator.java

```diff
@@ -416,23 +416,46 @@ public long getRetainedSizeInBytes()
 
         public void partitionPage(Page page)
         {
-            requireNonNull(page, "page is null");
+            if (requireNonNull(page, "page is null").getPositionCount() == 0) {
```

Comment:
Move the `requireNonNull` to a separate line. It muddles the condition and makes it harder to read.

Reply:
Moved